core/statistics/statistics.go

```diff
@@ -150,6 +157,7 @@ func (d *Data) SetEndBlockDuration(timeEnd time.Time, height uint64) {
 		defer d.BlockEnd.Unlock()
 		d.BlockEnd.Height.Set(float64(height))
 		d.BlockEnd.Duration.Set(timeEnd.Sub(d.BlockStart.time).Seconds())
+		d.BlockEnd.Timestamp.Set(float64(timeEnd.UnixNano()) / 1e9)
```

Comment:
You should use [Block.Header.Time](https://github.com/tendermint/tendermint/blob/78144306dd98157cffc17d896cdd01ea08d4d131/types/block.go#L328) instead own recorded timestamp, because it will lead to incorrect info during synchronization.